config/initializers/menus.rb

```diff
@@ -31,6 +31,18 @@
 require "redmine/menu_manager"
 
 Redmine::MenuManager.map :top_menu do |menu|
+  menu.push :portfolios,
+            { controller: "/portfolios", action: "index" },
+            context: :modules,
+            caption: I18n.t("label_portfolio_plural"),
+            icon: "briefcase",
+            if: ->(_) {
+              OpenProject::FeatureDecisions.portfolio_models_active? &&
+                (User.current.logged? || !Setting.login_required?) &&
```

Comment:
I think the checks in this line are not needed here since we are already checking for the user to have permissions.